[CI] GeoIpProcessorNonIngestNodeIT testLazyLoading failing

There have been some changes here around the introduction of a test fixture. Perhaps that is at play?

**Build scan**:
https://gradle-enterprise.elastic.co/s/efjcopyvrgc2q/tests/:modules:ingest-geoip:internalClusterTest/org.elasticsearch.ingest.geoip.GeoIpProcessorNonIngestNodeIT/testLazyLoading#1

**Repro line**:
`./gradlew ':modules:ingest-geoip:internalClusterTest' --tests "org.elasticsearch.ingest.geoip.GeoIpProcessorNonIngestNodeIT.testLazyLoading" -Dtests.seed=CC3286F211876D5A -Dtests.security.manager=true -Dtests.locale=sl-SI -Dtests.timezone=Greenwich -Druntime.java=11`

**Reproduces locally?**:
No

**Applicable branches**:
`master`

**Failure history**:
https://gradle-enterprise.elastic.co/scans/tests?search.relativeStartTime=P7D&search.timeZoneId=America/Los_Angeles&tests.container=org.elasticsearch.ingest.geoip.GeoIpProcessorNonIngestNodeIT&tests.sortField=FAILED&tests.test=testLazyLoading&tests.unstableOnly=true

**Failure excerpt**:
```
org.elasticsearch.ingest.geoip.GeoIpProcessorNonIngestNodeIT > testLazyLoading FAILED
    java.lang.AssertionError: Shard [.geoip_databases][0] is still locked after 5 sec waiting
        at __randomizedtesting.SeedInfo.seed([CC3286F211876D5A:A81B6736C4820283]:0)
        at org.junit.Assert.fail(Assert.java:88)
        at org.elasticsearch.test.InternalTestCluster.assertAfterTest(InternalTestCluster.java:2321)
        at org.elasticsearch.test.ESIntegTestCase.afterInternal(ESIntegTestCase.java:551)
        at org.elasticsearch.test.ESIntegTestCase.cleanUpCluster(ESIntegTestCase.java:2073)
```

--- a/modules/ingest-geoip/src/internalClusterTest/java/org/elasticsearch/ingest/geoip/AbstractGeoIpIT.java
+++ b/modules/ingest-geoip/src/internalClusterTest/java/org/elasticsearch/ingest/geoip/AbstractGeoIpIT.java
@@ -49,6 +49,7 @@ public abstract class AbstractGeoIpIT extends ESIntegTestCase {
         }
         return Settings.builder()
             .put("ingest.geoip.database_path", databasePath)
+            .put(GeoIpDownloaderTaskExecutor.ENABLED_SETTING.getKey(), false)
             .put(super.nodeSettings(nodeOrdinal))
             .build();
     }

--- a/modules/ingest-geoip/src/internalClusterTest/java/org/elasticsearch/ingest/geoip/GeoIpDownloaderIT.java
+++ b/modules/ingest-geoip/src/internalClusterTest/java/org/elasticsearch/ingest/geoip/GeoIpDownloaderIT.java
@@ -25,6 +25,7 @@ import org.elasticsearch.search.SearchHit;
 import org.elasticsearch.search.sort.SortOrder;
 import org.elasticsearch.test.ESIntegTestCase.ClusterScope;
 import org.elasticsearch.test.ESIntegTestCase.Scope;
+import org.junit.After;
 
 import java.io.BufferedOutputStream;
 import java.io.ByteArrayInputStream;
@@ -56,14 +57,21 @@ public class GeoIpDownloaderIT extends AbstractGeoIpIT {
 
     @Override
     protected Settings nodeSettings(int nodeOrdinal) {
-        Settings.Builder settings = Settings.builder()
-            .put(super.nodeSettings(nodeOrdinal))
-            .put(GeoIpDownloaderTaskExecutor.ENABLED_SETTING.getKey(), false);
+        Settings.Builder settings = Settings.builder().put(super.nodeSettings(nodeOrdinal));
         String endpoint = System.getProperty("geoip_endpoint");
         if (endpoint != null) {
             settings.put(GeoIpDownloader.ENDPOINT_SETTING.getKey(), endpoint);
         }
         return settings.build();
+    }
+
+    @After
+    public void disableDownloader(){
+        ClusterUpdateSettingsResponse settingsResponse = client().admin().cluster()
+            .prepareUpdateSettings()
+            .setPersistentSettings(Settings.builder().put(GeoIpDownloaderTaskExecutor.ENABLED_SETTING.getKey(), false))
+            .get();
+        assertTrue(settingsResponse.isAcknowledged());
     }
 
     public void testGeoIpDatabasesDownload() throws Exception {

